File: src/main/java/axirassa/services/runners/FeedbackAggregationServiceJob.java

package axirassa.services.runners;

import org.quartz.Job;
import org.quartz.JobDataMap;
import org.quartz.JobExecutionContext;
import org.quartz.JobExecutionException;

import axirassa.ioc.IocStandalone;
import axirassa.services.FeedbackAggregationService;
import axirassa.services.Service;

public class FeedbackAggregationServiceJob implements Job {

	public static final String DATABASE_SESSION = "session.database";
	public static final String MESSAGING_SESSION = "session.messaging";


	@Override
	public void execute (JobExecutionContext context) throws JobExecutionException {
		try {
			JobDataMap datamap = context.getJobDetail().getJobDataMap();
            // TODO use datamap to populate services for injection
            Service service = IocStandalone.autobuild(FeedbackAggregationService.class);
			service.execute();
		} catch (Throwable e) {
			throw new JobExecutionException(e);
		}
	}
}

File: src/main/java/axirassa/services/runners/FeedbackAggregationServiceJob.scala

package axirassa.services.runners

import axirassa.ioc.IocStandalone
import org.quartz.Job
import org.quartz.JobDataMap
import org.quartz.JobExecutionContext
import org.quartz.JobExecutionException

import axirassa.services.FeedbackAggregationService
import axirassa.services.Service

object FeedbackAggregationServiceJob {
    val DATABASE_SESSION = "session.database"
    val MESSAGING_SESSION = "session.messaging"
}

class FeedbackAggregationServiceJob extends Job {

    override def execute(context : JobExecutionContext) {
        try {
            val datamap = context.getJobDetail().getJobDataMap()
            val service = IocStandalone.autobuild(classOf[FeedbackAggregationService])
            service.execute()
        } catch {
            case e => throw new JobExecutionException(e)
        }
    }
}
